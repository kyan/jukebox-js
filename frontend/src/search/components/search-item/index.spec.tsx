import React from 'react'
import { mount } from 'enzyme'
import SearchItem from './index'

const validTrack = {
  name: 'track name',
  artist: {
    name: 'artist name'
  },
  album: {
    name: 'album name'
  },
  metrics: {},
  image: 'image1',
  explicit: false
}
const explicitTrack = {
  name: 'track name',
  artist: {
    name: 'artist name'
  },
  album: {
    name: 'album name'
  },
  metrics: {},
  image: 'image1',
  explicit: true
}

describe('SearchItem', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    const onAddMock = jest.fn().mockName('onAddMock')
    const onClickMock = jest.fn().mockName('onClickMock')

    test('with valid props it renders correctly', () => {
      const wrapper = mount(
        <SearchItem
          className={'item-class'}
          track={validTrack}
          onAdd={onAddMock}
          onClick={onClickMock}
        />
      )

      expect(wrapper).toMatchSnapshot()
      wrapper.find('Image').simulate('click')
      expect(onClickMock).toHaveBeenCalled()
      wrapper.find('.search-list-item__add').simulate('click')
      expect(onAddMock).toHaveBeenCalled()
    })

    test('with explicit track', () => {
      const wrapper = mount(
        <SearchItem
          className={'item-class'}
          track={explicitTrack}
          onAdd={onAddMock}
          onClick={onClickMock}
        />
      )

      expect(wrapper).toMatchSnapshot()
      wrapper.find('Image').simulate('click')
      expect(onClickMock).not.toHaveBeenCalled()
      wrapper.find('.search-list-item__add').simulate('click')
      expect(onAddMock).toHaveBeenCalled()
    })

    test('with no add handler', () => {
      const wrapper = mount(
        <SearchItem className={'item-class'} track={validTrack} onClick={onClickMock} />
      )

      expect(wrapper).toMatchSnapshot()
      wrapper.find('Image').simulate('click')
      expect(onClickMock).toHaveBeenCalled()
      expect(wrapper.find('.search-list-item__add')).toHaveLength(0)
    })
  })
})
