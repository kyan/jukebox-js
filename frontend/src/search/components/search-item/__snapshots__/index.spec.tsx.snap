// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem render with explicit track 1`] = `
<SearchItem
  className="item-class"
  onAdd={[MockFunction onAddMock]}
  onClick={[MockFunction onClickMock]}
  track={
    Object {
      "album": Object {
        "name": "album name",
      },
      "artist": Object {
        "name": "artist name",
      },
      "explicit": true,
      "image": "image1",
      "metrics": Object {},
      "name": "track name",
    }
  }
>
  <div
    className="item-class search-list-item disabled"
  >
    <Image
      as="img"
      className="search-list-item__image"
      disabled={true}
      floated="left"
      size="tiny"
      src="image1"
      title="Click to add - track name - artist name"
      ui={true}
    >
      <img
        className="ui tiny disabled left floated image search-list-item__image"
        src="image1"
        title="Click to add - track name - artist name"
      />
    </Image>
    <ListContent>
      <div
        className="content"
      >
        <div
          className="search-list-item__header"
        >
          track name
           - 
          artist name
        </div>
        <div
          className="search-list-item__content"
        >
          album name
        </div>
        <VoteInfo
          metrics={Object {}}
        >
          <VotedBy
            show={false}
            size="mini"
          />
        </VoteInfo>
        <span
          className="search-list-item__add"
          onClick={[MockFunction onAddMock]}
        >
          Add to mix
        </span>
      </div>
    </ListContent>
  </div>
</SearchItem>
`;

exports[`SearchItem render with no add handler 1`] = `
<SearchItem
  className="item-class"
  onClick={[MockFunction onClickMock]}
  track={
    Object {
      "album": Object {
        "name": "album name",
      },
      "artist": Object {
        "name": "artist name",
      },
      "explicit": false,
      "image": "image1",
      "metrics": Object {},
      "name": "track name",
    }
  }
>
  <div
    className="item-class search-list-item"
  >
    <Image
      as="img"
      className="search-list-item__image"
      disabled={false}
      floated="left"
      onClick={[MockFunction onClickMock]}
      size="tiny"
      src="image1"
      title="Click to add - track name - artist name"
      ui={true}
    >
      <img
        className="ui tiny left floated image search-list-item__image"
        onClick={[MockFunction onClickMock]}
        src="image1"
        title="Click to add - track name - artist name"
      />
    </Image>
    <ListContent>
      <div
        className="content"
      >
        <div
          className="search-list-item__header"
        >
          track name
           - 
          artist name
        </div>
        <div
          className="search-list-item__content"
        >
          album name
        </div>
        <VoteInfo
          metrics={Object {}}
        >
          <VotedBy
            show={false}
            size="mini"
          />
        </VoteInfo>
      </div>
    </ListContent>
  </div>
</SearchItem>
`;

exports[`SearchItem render with valid props it renders correctly 1`] = `
<SearchItem
  className="item-class"
  onAdd={[MockFunction onAddMock]}
  onClick={[MockFunction onClickMock]}
  track={
    Object {
      "album": Object {
        "name": "album name",
      },
      "artist": Object {
        "name": "artist name",
      },
      "explicit": false,
      "image": "image1",
      "metrics": Object {},
      "name": "track name",
    }
  }
>
  <div
    className="item-class search-list-item"
  >
    <Image
      as="img"
      className="search-list-item__image"
      disabled={false}
      floated="left"
      onClick={[MockFunction onClickMock]}
      size="tiny"
      src="image1"
      title="Click to add - track name - artist name"
      ui={true}
    >
      <img
        className="ui tiny left floated image search-list-item__image"
        onClick={[MockFunction onClickMock]}
        src="image1"
        title="Click to add - track name - artist name"
      />
    </Image>
    <ListContent>
      <div
        className="content"
      >
        <div
          className="search-list-item__header"
        >
          track name
           - 
          artist name
        </div>
        <div
          className="search-list-item__content"
        >
          album name
        </div>
        <VoteInfo
          metrics={Object {}}
        >
          <VotedBy
            show={false}
            size="mini"
          />
        </VoteInfo>
        <span
          className="search-list-item__add"
          onClick={[MockFunction onAddMock]}
        >
          Add to mix
        </span>
      </div>
    </ListContent>
  </div>
</SearchItem>
`;
